scripts/operations/ceph/enable-smart-mon-storage-nodes.sh

```diff
@@ -0,0 +1,113 @@
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+smartmon_url=''
+ssh_options="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+
+function get_smartmon_url() {
+  local repos
+
+  if ! curl -sSfk https://packages.local/service/rest/v1/repositories >&/dev/null; then
+    echo "***"
+    echo "WARNING: Unable to contact Nexus! This must be addressed before proceeding."
+    echo "***"
+    return 1
+  fi
+
+  # get a list of repos that start with "csm-noos"
+  if IFS=$'\n' read -rd '' -a repos; then
+        :
+  fi <<< "$(curl -sSfk https://packages.local/service/rest/v1/repositories | jq -r '.[] | .["name"]' | grep ^csm-noos | tr '\n' ' ')"
+
+  echo "Will look for smart-mon rpm in the following repos: ${repos[@]}"
+
+  # search through the csm-noos repos looking for our package
+  for repo in "${repos[@]}"; do
+    # Retrieve the packages from nexus
+    if [[ -z "$smartmon_url" ]]; then
+      smartmon_url=$(paginate "https://packages.local/service/rest/v1/components?repository=$repo" \
+        | jq -r '.items[] | .assets[] | .downloadUrl' | grep smart-mon | sort -V | tail -1)
+      if [[ $? -ne 0 ]]; then
+        return 1
+      fi
+    fi
+  done
+
+  if [[ -z  "$smartmon_url" ]]; then
+    echo WARNING: unable to install smart-mon rpm
+    return 1
+  fi
+}
+
+function paginate() {
+  local url="$1"
+  local token
+
+  if test -z $url; then
+    echo "ERROR: paginate() called without an argument"
+    return 1
+  fi
+
+  # check if last char of url is a space
+  if [[ "${url: -1}" == " " ]]; then
+    # remove last character if it is a space
+    url=${url::-1}
+  fi
+
+  { token="$(curl -sSk "$url" | tee /dev/fd/3 | jq -r '.continuationToken // null')"; } 3>&1
+
+  if test -z $token; then
+    echo "ERROR on line $LINENO: unable to retreive continuation token, exiting"
+    return 1
+  fi
+
+  until [[ $token == "null" ]]; do
+    {
+      token="$(curl -sSk "$url&continuationToken=${token}" | tee /dev/fd/3 | jq -r '.continuationToken // null')"
+    } 3>&1
+
+    if test -z $token; then
+      echo "ERROR on line $LINENO: unable to retreive continuation token, exiting"
+      return 1
+    fi
+  done
+}
+
+get_smartmon_url
+if [[ $? -ne 0 ]]; then
+  exit 1
+fi
```

Comment:
Check exit code directly, this will help when `set -e` is set. Otherwise if line 98 failed, the script would exit before running line 99.
```suggestion
if ! get_smartmon_url; then
  exit 1
fi
```